presto-thrift-tpch-node/src/main/java/com/facebook/presto/thrift/node/ThriftServerTpch.java

```diff
@@ -0,0 +1,462 @@
+    private final ListeningExecutorService splitsExecutor =
+            listeningDecorator(newCachedThreadPool(threadsNamed("splits-generator-%s")));
+    private final ListeningExecutorService dataExecutor =
+            listeningDecorator(newCachedThreadPool(threadsNamed("data-generator-%s")));
+    private final ListeningExecutorService indexDataExecutor =
+            listeningDecorator(newCachedThreadPool(threadsNamed("index-data-generator-%s")));
+    private final TpchIndexedData indexedData = new TpchIndexedData("tpchindexed", INDEX_SPEC);
+
+    @Override
+    public List<ThriftPropertyMetadata> listSessionProperties()
+    {
+        return ImmutableList.of(new ThriftPropertyMetadata(NUMBER_OF_SPLITS_PARAMETER, "integer", "Number of splits",
+                new ThriftSessionValue(false, null, 3, null, null, null),
+                false));
+    }
+
+    @Override
+    public List<String> listSchemaNames()
+    {
+        return SCHEMAS;
+    }
+
+    @Override
+    public List<ThriftSchemaTableName> listTables(@Nullable String schemaNameOrNull)
+    {
+        List<ThriftSchemaTableName> result = new ArrayList<>();
+        for (String schemaName : getSchemaNames(schemaNameOrNull)) {
+            for (TpchTable<?> tpchTable : TpchTable.getTables()) {
+                result.add(new ThriftSchemaTableName(schemaName, tpchTable.getTableName()));
+            }
+        }
+        return result;
+    }
+
+    private static List<String> getSchemaNames(String schemaNameOrNull)
+    {
+        if (schemaNameOrNull == null) {
+            return SCHEMAS;
+        }
+        else if (SCHEMAS.contains(schemaNameOrNull)) {
+            return ImmutableList.of(schemaNameOrNull);
+        }
+        else {
+            return ImmutableList.of();
+        }
+    }
+
+    @Override
+    public ThriftNullableTableMetadata getTableMetadata(ThriftSchemaTableName schemaTableName)
+    {
+        String schemaName = schemaTableName.getSchemaName();
+        String tableName = schemaTableName.getTableName();
+        if (!SCHEMAS.contains(schemaName) || TpchTable.getTables().stream().noneMatch(table -> table.getTableName().equals(tableName))) {
+            return new ThriftNullableTableMetadata(null);
+        }
+        TpchTable<?> tpchTable = TpchTable.getTable(schemaTableName.getTableName());
+        List<ThriftColumnMetadata> columns = new ArrayList<>();
+        for (TpchColumn<? extends TpchEntity> column : tpchTable.getColumns()) {
+            columns.add(new ThriftColumnMetadata(column.getSimplifiedColumnName(), getThriftType(column.getType()), null, false));
+        }
+        columns.add(new ThriftColumnMetadata(ROW_NUMBER_COLUMN_NAME, "bigint", null, true));
+        return new ThriftNullableTableMetadata(new ThriftTableMetadata(schemaTableName, columns));
+    }
+
+    private static String getThriftType(TpchColumnType tpchType)
+    {
+        return TpchMetadata.getPrestoType(tpchType).getTypeSignature().toString();
```

Comment:
I would put a comment saying that the thrift type for all the TPCH data happens to match the presto typesignature name